src/core/lib/event_engine/ares_resolver.cc

```diff
@@ -788,4 +789,38 @@ void (*event_engine_grpc_ares_test_only_inject_config)(ares_channel* channel) =
 
 bool g_event_engine_grpc_ares_test_only_force_tcp = false;
 
+bool ShouldUseAresDnsResolver() {
+#if defined(GRPC_POSIX_SOCKET_ARES_EV_DRIVER) || \
+    defined(GRPC_WINDOWS_SOCKET_ARES_EV_DRIVER)
+  auto resolver_env = grpc_core::ConfigVars::Get().DnsResolver();
+  return resolver_env.empty() || absl::EqualsIgnoreCase(resolver_env, "ares");
+#else   // defined(GRPC_POSIX_SOCKET_ARES_EV_DRIVER) ||
+        // defined(GRPC_WINDOWS_SOCKET_ARES_EV_DRIVER)
+  return false;
+#endif  // defined(GRPC_POSIX_SOCKET_ARES_EV_DRIVER) ||
+        // defined(GRPC_WINDOWS_SOCKET_ARES_EV_DRIVER)
+}
+
+absl::Status AresInit() {
```

Comment:
This is dead code since it's never used or tested. I think it's the right implementation, but it should either be used or deleted. I left a suggestion below.

Reply:
thanks, done.